rest/admin_api_test.go

```diff
@@ -34,6 +34,81 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// Reproduces CBG-1412 - JSON strings in some responses not being correctly escaped
+func TestPutDocSpecialChar(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+	testCases := []struct {
+		name         string
+		pathDocID    string
+		method       string
+		body         string
+		expectedResp int
+		eeOnly       bool
+	}{
+		{
+			name:         "Double quote PUT",
+			pathDocID:    `doc"55"`,
+			method:       "PUT",
+			body:         "{}",
+			expectedResp: http.StatusCreated,
+			eeOnly:       false,
+		},
+		{
+			name:         "Double quote PUT for replicator2",
+			pathDocID:    `doc"77"?replicator2=true`,
+			method:       "PUT",
+			body:         "{}",
+			expectedResp: http.StatusCreated,
+			eeOnly:       true,
+		},
+		{
+			name:         "Local double quote PUT",
+			pathDocID:    `_local/doc"57"`,
+			method:       "PUT",
+			body:         "{}",
+			expectedResp: http.StatusCreated,
+			eeOnly:       false,
+		},
+		{
+			name:         "Double quote PUT with attachment",
+			pathDocID:    `doc"59"/attachMe`,
+			method:       "PUT",
+			body:         "{}",
+			expectedResp: http.StatusCreated, // Admin Docs expected response http.StatusOK
+			eeOnly:       false,
+		},
```

Comment:
I see you had a test case for POST but it was removed.

Looks like handlePostDoc still has an unescaped docID field in the response body.
Is there a plan to include that in this PR?

Reply:
I didn't escape the DocID for handlePostDoc because when you try to use unescaped JSON, it returns HTTP status 400 (http.StatusBadRequest).

The test checked if the 400 error was returned for unescaped JSON to the POST doc endpoint and I didn't think that would be relevant to this ticket.

Reply:
Looks like handlePostDoc is already dealing with ensuring the doc ID is well-formed due to it coming out of the request body, I didn't notice that previously.

Looks good as is!